src/eml2_0/HdfProxyMPI.cpp

```diff
@@ -90,8 +90,27 @@ void HdfProxyMPI::open()
 		throw invalid_argument("The HDF5 permission access is unknown.");
 	}
 
+	dsetPlistId.reset(new hid_t);
+	*dsetPlistId = H5Pcreate(H5P_DATASET_XFER);
+
 	/* Release file-access template */
 	if (H5Pclose(fapl_id)) {
 		throw invalid_argument("Cannot release the parallel file-access template");
 	}
 }
+
+void HdfProxyMPI::setCollectiveIO()
+{
+	if( H5Pset_dxpl_mpio( *dsetPlistId, H5FD_MPIO_COLLECTIVE) < 0)
+	{
+		throw invalid_argument("Can not set the collective IO flag");
+	}
+}
+
+void HdfProxyMPI::setIndependentIO()
+{
+	if( H5Pset_dxpl_mpio( *dsetPlistId, H5FD_MPIO_INDEPENDENT) < 0)
```

Comment:
A check on the existency of dsetPlistId should be done in case someone call this method before open().